src/crimson/os/seastore/seastore_types.h

```diff
@@ -1237,19 +1279,44 @@ struct __attribute__((packed)) root_t {
   }
 };
 
+struct alloc_blk_t {
+  alloc_blk_t(
+    paddr_t paddr,
+    laddr_t laddr,
+    extent_len_t len,
+    extent_types_t type)
+    : paddr(paddr), laddr(laddr), len(len), type(type)
+  {}
+
+  explicit alloc_blk_t() = default;
+
+  paddr_t paddr = P_ADDR_NULL;
+  laddr_t laddr = L_ADDR_NULL;
+  extent_len_t len = 0;
+  extent_types_t type = extent_types_t::ROOT;
+  DENC(alloc_blk_t, v, p) {
+    DENC_START(1, 1, p);
+    denc(v.paddr, p);
+    denc(v.laddr, p);
+    denc(v.len, p);
+    denc(v.type, p);
+    DENC_FINISH(p);
+  }
+};
+
 // use absolute address
-struct rbm_alloc_delta_t {
+struct alloc_delta_t {
```

Comment:
Commit that introduces this has a typo:

    crimson/os/seastore: rename rbm_alloc_delta_t to all_delta_t
->

    crimson/os/seastore: rename rbm_alloc_delta_t to alloc_delta_t